components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2106,4 +2272,173 @@ private void handleInternalTokenRevocation(String consumerKey, Properties proper
+                JsonNode redirectURIArray = mapper.readTree(uri.toURL());
+                if (redirectURIArray.isArray()) {
+                    Iterator<JsonNode> itr = redirectURIArray.iterator();
+                    while (itr.hasNext()) {
+                        JsonNode item = itr.next();
+                        fetchedRedirectURI.add(item.asText());
+                    }
+                }
+                if (!fetchedRedirectURI.containsAll(redirectURIs)) {
+                    throw handleClientError(INVALID_REQUEST, "Redirect URI missing in sector " +
+                            "identifier URI set");
+                }
+            } catch (IOException e) {
+                throw handleClientError(INVALID_REQUEST, "Invalid sector identifier URI");
+            }
+        }
+    }
+
+    /**
+     * Get call back URIs as a list
+     * @param application  OAuthConsumerAppDTO
+     * @return list of callback urls
+     */
+    private List<String> getRedirectURIList(OAuthConsumerAppDTO application) {
+
+        List<String> callBackURIList = new ArrayList<>();
+        // Need to split the redirect uris for validating the host names since it is combined
+        // into one regular expression.
+        if (application.getCallbackUrl().startsWith(
+                OAuthConstants.CALLBACK_URL_REGEXP_PREFIX)) {
+            String redirectURI = application.getCallbackUrl();
+            redirectURI = redirectURI.substring(redirectURI.indexOf("(") + 1,
+                    redirectURI.indexOf(")"));
+            callBackURIList = Arrays.asList(redirectURI.split("\\|"));
+        }
+        return callBackURIList;
+    }
+
+    /**
+     * filter allowed signature algorithms
+     * @param algorithm algorithm to be validated
+     * @param configName configuration to read allowed algorithms
+     * @throws IdentityOAuthClientException if the algorithm is not allowed
+     */
+    public void filterSignatureAlgorithms(String algorithm, String configName) throws IdentityOAuthClientException {
+
+        List<String> allowedSignatureAlgorithms = IdentityUtil.getPropertyAsList(configName);
+        if (!(allowedSignatureAlgorithms.isEmpty())) {
+            if (!allowedSignatureAlgorithms.contains(algorithm)) {
+                String msg = String.format("'%s' Signing Algorithm is not allowed.", algorithm);
+                throw handleClientError(INVALID_REQUEST, msg);
+            }
+        }
+    }
+
+    /**
+     * filter allowed token endpoint authentication methods
+     * @param authMethod   authentication method to be validated
+     * @throws IdentityOAuthClientException if the token endpoint authentication method is not allowed
+     */
+    public void filterTokenEndpointAuthMethods(String authMethod) throws IdentityOAuthClientException {
+
+        List<String> authMethods = Arrays.asList(OAuth2Util.getSupportedClientAuthMethods());
+        if (!(authMethods.isEmpty())) {
+            if (!authMethods.contains(authMethod)) {
```

Comment:
we can omit the isEmpty check as we are are doing a contains check. Lets make this change in other places as well.

Also can Arrays.asList produce nulls? if so add a null check.

Reply:
getSupportedClientAuthMethods will not return null